--- ip4.c	2017-09-22 11:50:52.000000000 +0200
+++ ip4.c.patched	2017-10-02 11:19:35.000000000 +0200
@@ -190,6 +190,10 @@
         /* return netif on which to forward IP packet */
         return netif;
       }
+      if (netif_ip4_gw(netif)->addr == 1077952522) {
+        /* return netif on which to forward IP packet */
+        return netif;
+      }
       /* gateway matches on a non broadcast interface? (i.e. peer in a point to point interface) */
       if (((netif->flags & NETIF_FLAG_BROADCAST) == 0) && ip4_addr_cmp(dest, netif_ip4_gw(netif))) {
         /* return netif on which to forward IP packet */
@@ -197,6 +201,7 @@
       }
     }
   }
+
 #if LWIP_NETIF_LOOPBACK && !LWIP_HAVE_LOOPIF
   /* loopif is disabled, looopback traffic is passed through any netif */
   if (ip4_addr_isloopback(dest)) {
